code/modules/ventcrawl/ventcrawl.dm

```diff
@@ -0,0 +1,223 @@
+var/list/ventcrawl_machinery = list(
+	/obj/machinery/atmospherics/unary/vent_pump,
+	/obj/machinery/atmospherics/unary/vent_scrubber
+	)
+
+// Vent crawling whitelisted items, whoo
+/mob/living/var/list/can_enter_vent_with = list(
+	/obj/item/weapon/implant,
+	/obj/item/device/radio/borg,
+	/obj/item/weapon/holder,
+	/obj/machinery/camera,
+	/mob/living/simple_animal/borer
+	)
+
+/mob/living/var/list/icon/pipes_shown = list()
+/mob/living/var/last_played_vent
+/mob/living/var/is_ventcrawling = 0
+/mob/var/next_play_vent = 0
+
+/mob/living/proc/can_ventcrawl()
+	return 0
+
+/mob/living/Login()
+	. = ..()
+	//login during ventcrawl
+	if(is_ventcrawling && istype(loc, /obj/machinery/atmospherics)) //attach us back into the pipes
+		remove_ventcrawl()
+		add_ventcrawl(loc)
+
+/mob/living/carbon/slime/can_ventcrawl()
+	if(Victim)
+		src << "<span class='warning'>You cannot ventcrawl while feeding.</span>"
+		return 0
+	return 1
+
+/mob/living/proc/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	for(var/type in can_enter_vent_with)
+		if(istype(carried_item, can_enter_vent_with))
+			return get_inventory_slot(carried_item) == 0
+	return 0
+
+/mob/living/carbon/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	if(carried_item in internal_organs)
+		return 1
+	return ..()
+
+/mob/living/carbon/human/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	if(carried_item in organs)
+		return 1
+	return ..()
+
+/mob/living/simple_animal/spiderbot/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	if(carried_item == held_item)
+		return 1
+	return ..()
+
+/mob/living/proc/ventcrawl_carry()
+	for(var/atom/A in src.contents)
+		if(!is_allowed_vent_crawl_item(A))
+			src << "<span class='warning'>You can't be carrying that when vent crawling!</span>"
+			return 0
+	return 1
+
+/mob/living/AltClickOn(var/atom/A)
```

Comment:
Is there a way to flip this around? Meaning, is there a proc which is fired on an object when someone alt-clicks on it? That way, you would define for both scrubbers and vents. Though, hrm, that'd be duplicate definitions, effectively. Hrm.